### PR TITLE
Bug 1905074: Added gherkin linter

### PR DESCRIPTION
**Pull request by makambalaji**
To maintain the gherkin standards, included the linter in package.json

### CODE REVIEW

**Comment by andrewballantyne** [comment]
@makambalaji, as discussed in our UI sync, did we want this part of the Frontend CI? It's not going to overly useful if we don't have PRs blocked by it. 🤔 

Probably should be careful not to add much extra time to CIs but I think as long as it's not part of `e2e-gcp-console`, it won't matter. 

cc @spadgett @dtaylor113 Thoughts?

**Comment by spadgett** [comment]
@andrewballantyne I agree. If we add a linter, we should run it as part of CI.

File: frontend/packages/dev-console/integration-tests/README.md [hunk 1]
```diff
@@ -79,3 +84,4 @@ frontend/packages/dev-console/integration-tests/
 1. Don't use static sleep statements (browser.sleep)
 2. Comments should be included wherever required
 3. Don't inlcude console.log statements while raising PR
+4. .gherkin-lintrc  configuration file present in forntend folder is used to set Gherkin standards
```
Comment (dtaylor113) [inline frontend/packages/dev-console/integration-tests/README.md:87]: Spelling: `configuration file present in frontend...`

**Review by dtaylor113 — CHANGES_REQUESTED** [review]
Looks good, just one small spelling/typo.

**Comment by makambalaji** [comment]
> @andrewballantyne I agree. If we add a linter, we should run it as part of CI.

@andrewballantyne  - It won't be linked with CI for now, QE team will execute locally.
In future, If multiple teams are planning to use this BDD approach,, then I recommend it to integrate with CI :-).
cc: @spadgett @dtaylor113

File: frontend/packages/dev-console/integration-tests/README.md (hunk 1 above)
Comment (makambalaji) [reply frontend/packages/dev-console/integration-tests/README.md]: updated

**Review by dtaylor113 — APPROVED** [review]
/lgtm

**Review by dtaylor113 — APPROVED** [review]
/lgtm

**Comment by makambalaji** [comment]
@andrewballantyne , @dtaylor113  - Can you please approve this pr

File: frontend/package.json [hunk 2]
```diff
@@ -18,6 +18,7 @@
     "coverage": "jest --coverage .",
     "eslint": "eslint --ext .js,.jsx,.ts,.tsx,.json,.gql,.graphql --color",
     "lint": "NODE_OPTIONS=--max-old-space-size=4096 yarn eslint .",
+    "gherkin-lint": "./node_modules/.bin/gherkin-lint -c ./.gherkin-lintrc ./packages/dev-console/integration-tests/features",
```
Comment (andrewballantyne) [inline frontend/package.json:21]: Your readme does not specify to run this command. Perhaps put it under [Generic Standards](https://github.com/openshift/console/blob/e0158df322c995c0cd153423a69c5743ba3f708d/frontend/packages/dev-console/integration-tests/README.md#generic-standards)?

Comment (makambalaji) [reply frontend/package.json:21]: updated

File: frontend/packages/dev-console/integration-tests/README.md [hunk 3]
```diff
@@ -79,3 +84,5 @@ frontend/packages/dev-console/integration-tests/
 1. Don't use static sleep statements (browser.sleep)
 2. Comments should be included wherever required
 3. Don't inlcude console.log statements while raising PR
+4. .gherkin-lintrc configuration file present in frontend folder is used to set Gherkin standards
+5. Execute the "npm run gherkin-lint" command for every QE pr
```
Comment (andrewballantyne) [inline frontend/packages/dev-console/integration-tests/README.md:88]: We use `yarn` throughout the READMEs, recommend you don't refer to `npm` (despite it being a dependency of `yarn`.

Comment (makambalaji) [reply frontend/packages/dev-console/integration-tests/README.md:88]: updated

**Review by andrewballantyne — APPROVED** [review]
/lgtm

**Comment by andrewballantyne** [comment]
/retest

PR merged and kicked the remaining PRs back through. Flake (JS Memory).

**Pull request merged**